src/cli.js

```diff
@@ -234,17 +234,24 @@ function cli(inputArgs) {
 
     var args = nopt(knownOpts, shortHands, inputArgs);
 
-    if (args.version) {
-        var cliVersion = require('../package').version;
+    var cliVersion = require('../package').version;
+    // TODO: Use semver.prerelease when it gets released
+    var usingPrerelease = /-nightly|-dev$/.exec(cliVersion);
+    if (args.version || usingPrerelease) {
         var libVersion = require('cordova-lib/package').version;
         var toPrint = cliVersion;
         if (cliVersion != libVersion || /-dev$/.exec(libVersion)) {
```

Comment:
`nightly` versions include SHA so they should be different. 
Although it might make sense just to use `usingPrerelease` flag in this check (there is a little probability of SHAs match).